src/sentry/monitors/models.py

```diff
@@ -481,21 +489,46 @@ def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
         if not affected:
             return False
 
-        event_manager = EventManager(
-            {
-                "logentry": {"message": f"Monitor failure: {self.monitor.name} ({reason})"},
-                "contexts": {"monitor": get_monitor_environment_context(self)},
-                "fingerprint": ["monitor", str(self.monitor.guid), reason],
+        failure_occurrence = {
+            "event": {
                 "environment": self.environment.name,
-                # TODO: Both of these values should be get transformed from context to tags
-                # We should understand why that is not happening and remove these when it correctly is
-                "tags": {"monitor.id": str(self.monitor.guid), "monitor.slug": self.monitor.slug},
+                "event_id": uuid.uuid4().hex,
+                "platform": "crons",
+                "project_id": self.monitor.project_id,
+                "received": timezone.now(),
+                "sdk": None,
+                "tags": {
+                    "monitor.id": str(self.monitor.guid),
+                    "monitor.slug": self.monitor.slug,
+                },
+                "timestamp": timezone.now(),
             },
-            project=Project(id=self.monitor.project_id),
+            "issue_title": f"Monitor failure: {self.monitor.name} ({reason})",
+            "subtitle": f"{self.environment.name}",
+            "resource_id": None,
+            "evidence_data": {"Timestamp": last_checkin},
+            "evidence_display": [
+                {"name": "Failure reason", "value": reason, "important": True},
+                {"name": "Environment", "value": self.environment.name, "important": False},
+            ],
+            "detection_time": timezone.now(),
+            "id": uuid.uuid4().hex,
+            "project_id": self.monitor.project_id,
+            "fingerprint": [f"monitor-{str(self.monitor.guid)}-{reason}"],
+        }
+        failure_occurrence["type"], failure_occurrence["level"] = get_group_type_and_level(reason)
+
+        topic = settings.KAFKA_INGEST_OCCURRENCES
+        cluster_name = settings.KAFKA_TOPICS[topic]["cluster"]
+        cluster_options = get_kafka_producer_cluster_options(cluster_name)
+        producer = Producer(cluster_options)
+
+        producer.produce(
+            topic=topic,
+            key=None,
+            value=json.dumps(failure_occurrence, default=str),
         )
-        event_manager.normalize()
-        data = event_manager.get_data()
-        insert_data_to_database_legacy(data)
```

Comment:
Might make sense to feature flag this so you can switch between creating issue through issue platform versus the old way.